compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -524,111 +525,113 @@ createProjectPackageDb opts fps = do
-                              , pkgId
-                              , pkgRefs)
-              let pkgIdsTopoSorted = reverse $ topSort depGraph
-              dbPathAbs <- makeAbsolute dbPath
-              projectPackageDatabaseAbs <- makeAbsolute projectPackageDatabase
-              forM_ pkgIdsTopoSorted $ \vertex -> do
-                  let ((src, templInstSrc, uid, dalf, bs), pkgId, _) =
-                          vertexToNode vertex
-                  when (uid /= primUnitId) $ do
-                      let unitIdStr = unitIdString uid
-                      let instancesUnitIdStr = "instances-" <> unitIdStr
-                      let pkgIdStr = T.unpack $ LF.unPackageId pkgId
-                      let (pkgName, mbPkgVersion) =
-                              fromMaybe (unitIdStr, Nothing) $ do
-                                  (uId, ver) <- stripInfixEnd "-" unitIdStr
-                                  guard $ all (`elem` '.' : ['0' .. '9']) ver
-                                  Just (uId, Just ver)
-                      let deps =
-                              [ unitIdString uId <.> "dalf"
-                              | ((_src, _templSrc, uId, _dalf, _bs), pId, _) <-
-                                    map vertexToNode $ reachable depGraph vertex
-                              , pkgId /= pId
-                              ]
-                      let workDir = dbPath </> unitIdStr <> "-" <> pkgIdStr
-                      createDirectoryIfMissing True workDir
-                      -- write the dalf package
-                      B.writeFile (workDir </> unitIdStr <.> "dalf") bs
-                      generateAndInstallIfaceFiles
-                          dalf
-                          src
-                          opts
-                          workDir
-                          dbPath
-                          projectPackageDatabase
-                          unitIdStr
-                          pkgIdStr
-                          pkgName
-                          mbPkgVersion
-                          deps
-
-                      unless (null templInstSrc) $
-                          generateAndInstallInstancesPkg
-                              templInstSrc
-                              opts
-                              dbPathAbs
-                              projectPackageDatabaseAbs
-                              unitIdStr
-                              instancesUnitIdStr
-                              pkgName
-                              mbPkgVersion
-                              deps
-
-       | length uniqSdkVersions <= 1 -> forM_ dars $
-            \ExtractedDar{..} -> installDar dbPath edConfFiles edDalfs edSrcs
-       | otherwise ->
+    deps <- handleSdkPackages $ filter (`notElem` basePackages) deps0
+    depsExtracted <- mapM extractDar deps
+    let uniqSdkVersions = nubSort $ filter (/= "0.0.0") $ thisSdkVer : map edSdkVersions depsExtracted
```

Comment:
Why do we filter out `0.0.0`? The only case where I can see this being useful is immediately after a release where things are compatible for a very short amount of time. But that doesn’t seem like something anyone should rely on so I’d rather prevent it.

Reply:
It's there because otherwise the assistant integration tests will fail, when they import SDK packages (which have a different sdk version than 0.0.0).

Reply:
I added a comment, I'd like to drop it if somehow possible.